### PR TITLE
Quick fix for issue with 2328

### PR DESCRIPTION
**Pull request by lucas-wilkins**
Quick fix to braking change on main from 2328

@krzywon @wpotrzebowski

### CODE REVIEW

**Comment by lucas-wilkins** [comment]
This is still failing for some reason

File: src/sas/system/config/config_meta.py [hunk 1]
```diff
@@ -205,17 +205,37 @@ def _generate_schema(self) -> Dict[str, SchemaElement]:
...
+
+        # This section deals with config values themselves
         if getattr(self, "_locked", False):
             try:
                 super().__setattr__(key, self._schema[key].coerce(value))
             except CoercionError:
                 raise TypeError(f"Tried to set bad value '{value}' to config entry of type '{self._schema[key]}'")
         else:
+
+            if key not in self.__dict__:
```
Comment (krzywon) [inline src/sas/system/config/config_meta.py:234]: The config isn't locked at this point, but you're throwing a ConfigLocked exception if the key isn't already set.

**Review by krzywon — APPROVED** [review]
Tested locally, unit tests now pass, and SasView launches. I have one suggestion, but it isn't a show-stopper.

File: src/sas/system/config/config_meta.py [hunk 2]
```diff
@@ -205,17 +205,45 @@ def _generate_schema(self) -> Dict[str, SchemaElement]:
...
+
+        # This section deals with control variables for the config
+
+        if not hasattr(self, "_meta_attributes"):
+
+            # The class is not set up at this point, don't do any checks
             super().__setattr__(key, value)
+            return
+
+        else:
```
[the remark this answers is not in the document]
Comment (lucas-wilkins) [reply src/sas/system/config/config_meta.py]: That's better. I think the logic can be improved further.

**Comment by lucas-wilkins** [comment]
@krzywon @wpotrzebowski Assuming the latest change hasn't broken anything again, I'll merge this before I go home today.

**Pull request merged**